<html>

<head>
<title>rTRMui: documentation</title>
</head>

<body>

<div style="text-align:justify;margin-bottom:100px">

<!--<h3 style="background-image:url(../pic/logo.png);background-size:175px;background-repeat:no-repeat;padding-left:175px">rTRMui: a graphical user interface for the identification of transcriptional regulatory modules</h3>-->

<!--<iframe src="http://localhost:8100" style="border: 1px solid #CCC; width: 440px; height: 500px"></iframe>-->

<h3 id="intro">Introduction</h3>

<div style="float:left;padding:5px;margin-right:10px;border-style:solid;border-width:1px;border-color:#AAAAAA"><img src="pic/logo.png" style="width:250px"><br>The rTRMui logo is an example of a TRM.</div><p>The goal of rTRMui is to identify transcription factor modules (TRMs), that is, complexes of transcription factors (see figure) involved in the regulation of gene expression in a particular cell type or condition. Our approach is centered around a target transcription factor (red circle in figure), for which there is experimental evidence of binding (e.g. by ChIP-seq) and some query transcription factors (blue circles in figure), which have binding sites around the peaks for the target transcription factor. We integrate this information with gene expression in the cell type and with protein-protein interaction data, and use it to predict TRMs.</p>

<p>To have a glance at rTRMui capabilities download the example session dataset and follow the example shown in the Tutorial. Below follows a detailed explanation of the different parameters and options supported in rTRMui.</p>


<h3 id="data_input">Data input</h3>
<p>
<h4>Target organism</h4>

The species where the ChIP-seq or alternative experimental technique has been performed for the target transcription factor.

<h4>Target transcription factor</h4>

The transcription factor for which a ChIP-seq experiment is available and around which we wish to identify TRMs. Search in the "Transcription factors" tab for the symbol of the transcription factor you want to use.

<h4>Query transcription factor</h4>

The list of transcription factors that we want to test whether they belong to the TRM around the target transcription factor. This list may come from any source of evidence. A typical case is the enrichment of transcription factor binding motifs in the binding regions of the target transcription factor. A file with each identifier followed by a new line must be specified. Supporting identifiers include PWM ids from the rTRM database or from the MotifDb package, as well as Entrezgene identifiers.

<h4>Expressed genes</h4>

List of expressed genes in the tissue of interestes. Typically this means the same tissue or cell type for which the ChIP-seq experiment for the target transcription factor was performed. The list must currently be specified as Entrezgene identifiers.

</p>

<h3 id="network_parameters">Network parameters</h3>
<p>
<h4>Extended TRM</h4>

This parameter controls whether the TRM is computed applying the maximum distance constraint to the distance between query and target transcription factors or if query transcription factors that satisfy the constraint when compared to other query transcription factors are also kept (FALSE by default).

<h4>Strict TRM</h4>

Whether a connected PPI module is enforced. If TRUE then only the connected component containing the target transcription factor is returned. If FALSE then it will return all the components (TRUE by default).

<h4>Bridge distance</h4>

Maximum distance between the query and the target transcription factors. If "Extended TRM" is FALSE then it is also the maximum distance between the query transcription factors.

<h4>Filter Ubiquitin/Sumo from PPI</h4>

Whether to remove from the PPI network the nodes corresponding to Ubiquitin and Sumo peptides (TRUE by default).

</p>
<h3 id="plot_parameters">Plot parameters</h3>

<p>

This section enable (basic) adjuting some of the plotting parameters that control the appearance of the TRM.

<h4>Margin</h4>

Controls the margin around the TRM to avoid nodes to get cropped when the node size is large.

<h4>Network layout</h4>

Selects the network layout for the TRM. Five layouts are available:

<ul>
	<il><strong>concentric:</strong> (default) a concentric layout with target TFs in the center, query TFs in the outer circle and bridge proteins in the middle circle.</il><br>
	<il><strong>arc:</strong> layout mimiquing the rTRM logo, i.e. a linear layout with the target TFs in the middle and the query TFs at each side, with the bridge proteins in between of both in the right side.</il><br>
	<il><strong>circle:</strong> circle layout from the igraph package.</il><br>
	<il><strong>kamada.kawai:</strong> from the igraph package.</il><br>
	<il><strong>fruchterman.reingold:</strong> from the igraph package.</il><br>
</ul>

<h4>Sort nodes by name</h4>

Controls whether nodes should be ordered by node name in "concentric" layout.

<h4>Node size</h4>

Controls node size.

<h4>Line size</h4>

Controls size of edges and node border.

<h4>Label size</h4>

Controls label size.

</p>

<h3 id="download">Download</h3>

<p>
	Three buttons with the labels "Plot", "Legend" and "Table" appear when a TRM has been successfully identified. Clicking in any of these buttons will download a PDF version of the TRM, a PDF version of the legend and a tab delimited text file with the information about TRMs found in the "Table" tab.
</p>

</div>
</body>
</html>
